Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,399 @@
+            /// the default value is sufficient.
+            /// For more information, see SplineUtility.GetNearestPoint.
+            /// </summary>
+            [Tooltip("The nearest point is calculated by finding the nearest point on the entire length of the spline " +
+                "using SearchResolution to divide into equally spaced line segments. Successive iterations will then " +
+                "subdivide further the nearest segment, producing more accurate results. In most cases, the default value (2) is sufficient.")]
+            public int m_SearchIteration;
+
+            /// <summary>Constructor with specific field values</summary>
+            /// <param name="enabled">Whether to enable automatic dolly</param>
+            /// <param name="positionOffset">Offset, in current position units, from the closest point on the path to the follow target</param>
+            /// <param name="searchResolution">We search between waypoints by dividing the segment into this many straight pieces</param>
+            /// <param name="searchIteration">Straight pieces defined by searchResolution are further subdivided.</param>
+            public AutoDolly(bool enabled, float positionOffset, int searchResolution, int searchIteration)
+            {
+                m_Enabled = enabled;
+                m_PositionOffset = positionOffset;
+                m_SearchResolution = searchResolution;
+                m_SearchIteration = searchIteration;
+            }
+        };
+
+        /// <summary>Controls how automatic dollying occurs</summary>
+        [Tooltip("Controls how automatic dollying occurs.  A Follow target is necessary to use this feature.")]
+        public AutoDolly m_AutoDolly = new AutoDolly(false, 0, 4, 2);
+
+        /// <summary>True if component is enabled and has a path</summary>
+        public override bool IsValid { get { return enabled && m_Track != null; } }
+
+        /// <summary>Get the Cinemachine Pipeline stage that this component implements.
+        /// Always returns the Body stage</summary>
+        public override CinemachineCore.Stage Stage { get { return CinemachineCore.Stage.Body; } }
+
+        /// <summary>
+        /// Report maximum damping time needed for this component.
+        /// </summary>
+        /// <returns>Highest damping setting in this component</returns>
+        public override float GetMaxDampTime() 
+        { 
+            var d2 = AngularDamping;
+            var a = Mathf.Max(m_Damping.x, Mathf.Max(m_Damping.y, m_Damping.z)); 
+            var b = Mathf.Max(d2.x, Mathf.Max(d2.y, d2.z)); 
+            return Mathf.Max(a, b); 
+        }
+        
+        /// <summary>
+        /// Subscribe to onSplineChanged if you'd like to react to changes to the Spline attached to this vcam.
+        /// This action is invoked by the Spline's changed event when a spline property is modified.
```

Comment:
Is this editor-only or does it also work in builds?

Reply:
Good question. I assumed it was not editor only, because the spline docs did not say it.  I checked and it is editor only. So I updated our docs, and pushed a PR to spline team too :)